src/drivers/uavcan/sensors/rangefinder.cpp

```diff
@@ -108,14 +108,13 @@ void UavcanRangefinderBridge::range_sub_cb(const
 		_inited = true;
 	}
 
-	/*
-	 * FIXME HACK
-	 * This code used to rely on msg.getMonotonicTimestamp().toUSec() instead of HRT.
-	 * It stopped working when the time sync feature has been introduced, because it caused libuavcan
-	 * to use an independent time source (based on hardware TIM5) instead of HRT.
-	 * The proper solution is to be developed.
-	 */
-	rangefinder->update(hrt_absolute_time(), msg.range);
+	int8_t quality = -1;
+
+	if (msg.reading_type == uavcan::equipment::range_sensor::Measurement::READING_TYPE_VALID_RANGE) {
```

Comment:
Seems like we could use more the enum? https://github.com/dronecan/DSDL/blob/f618adf88990829abd152b48a7bbdfd58881a5a6/uavcan/equipment/range_sensor/1050.Measurement.uavcan#L19-L22

Reply:
For example we could use READING_TYPE_TOO_CLOSE and READING_TYPE_TOO_FAR properly, then lift or remove UAVCAN_RNG_MIN/UAVCAN_RNG_MAX so they don't need to be manually configured.

Reply:
I'm not sure we have the ability to tell too close or too far from all the distance sensor drivers.